### ISSUE TITLE
support for java 8 Date-Time classes

### ISSUE THREAD

**Issue by bassmake** [cucumber/cucumber-jvm#974]
It would be nice to have support for Java 8 Date Time classes in cucumber.
in scenario:

```
And valid from is set to '2016-04-08T12:30:00'
```

in java:

```
    Given("valid from is set to '(.*)'", (LocalDateTime validFrom) -> {
      templateBuilder.validFrom(validFrom);
    });
```

using `LocalDateTime.parse` should be enough for this

**Comment by stanislav-antufeev** [comment]
How are you going yo specify format of your string date in step definition ?

**Comment by bassmake** [comment]
something similar to https://github.com/cucumber/cucumber-jvm/issues/84

**Comment by mpkorstanje** [comment]
Currently cucumber is still at java 1.7 so it won't be able to support the date classes from Java 8 out of the box. This Stackoverflow post may however help for the time being:

https://stackoverflow.com/questions/49565284/cucumber-not-serializing-date-string-from-datatable-in-feature-file-to-a-localda

Note that once #1248 is done you'll be able to use a custom parameter type to define your own dates.
